akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ActiveActiveShardingReplication.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.eventstream.EventStream
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.ApiMayChange
+import akka.annotation.DoNotInherit
+import akka.annotation.InternalApi
+import akka.persistence.typed.PublishedEvent
+
+import scala.collection.JavaConverters._
+
+/**
+ * Used when sharding Active Active entities in multiple instances of sharding, for example one per DC in a Multi DC
+ * Akka Cluster.
+ *
+ * This actor should be started once on each node where Active Active entities will run (the same nodes that you start
+ * sharding on).
+ *
+ * Subscribes to locally written events through the event stream and sends the seen events to all the sharded replicas
+ * which can then fast forward their cross-replica event streams to improve latency but still allowing slower poll
+ * frequency for the cross replica queries. Note that since message delivery is at-most-once this can not be the only
+ * channel for replica events - the entities must still tail events from the journals of other replicas.
+ *
+ * The events are forwarded as [[akka.cluster.sharding.typed.ShardingEnvelope]] this will work out of the box if the the
```

Comment:
so the user can't have an entity/shard extractor right? It wouldn't type correctly

Reply:
Actually it seems as though we always wrap a user specified extractor in `ExtractorAdapter` when we start typed sharding, so this comment is wrong, it will always automagically work.

Reply:
Great!